mod-source-record-manager-server/src/main/java/org/folio/services/ChangeEngineServiceImpl.java

```diff
@@ -179,14 +177,16 @@ private void addAdditionalFieldsToMarcRecord(Record record) {
     Future<List<Record>> future = Future.future();
     try {
       SourceStorageBatchClient client = new SourceStorageBatchClient(params.getOkapiUrl(), params.getTenantId(), params.getToken());
+
       RecordCollection recordCollection = new RecordCollection()
         .withRecords(parsedRecords)
         .withTotalRecords(parsedRecords.size());
+
       client.postSourceStorageBatchRecords(recordCollection, response -> {
-        if (response.statusCode() == HttpStatus.HTTP_CREATED.toInt()) {
-          future.complete(parsedRecords);
+        if (isStatus(response, HTTP_CREATED) || isPartialSuccess(response)) {
+          response.bodyHandler(it -> future.complete(it.toJsonObject().mapTo(RecordCollection.class).getRecords()));
```

Comment:
"it.toJsonObject().mapTo(RecordCollection.class)"
if there will be na error future will be never ended

Reply:
done